### ISSUE TITLE
Arsenal Doesn't Save/Restore Gunbag Weapons

### ISSUE THREAD

**Issue by jokoho48** [acemod/ACE3#6211]
**Arma 3 Version:** `1.80` (stable / rc / dev)
**CBA Version:** `3.6.1.180131` (stable / dev + commit hash)
**ACE3 Version:** `3.12.1` (stable / dev + commit hash)

**Mods:**
```
- CBA_A3
- ace
```

**Description:**
- The ACE Arsenal Does not Save and Restores Weapons that are stored in gun bags

**Steps to reproduce:**
- Take Gunbag and put in a Weapon
- Save Loadout
- Load Loadout
- Weapon Gone

**Comment by dedmen** [comment]
Variable is stored on the gunbag. 
We could definitely just store all variables on backpacks. But IMO that's a bad Idea. Let mods configure what variable names should be stored and then store them in the loadout array.
Maybe use CBA hashes to get a good storage layout?

Could also be useful for TFAR/ACRE. They have special code to keep variables on their backpacks when you enter/leave Arsenal. But would also be nice if they could be stored. However TFAR needs to manipulate the data before it is stored or after it is loaded. I don't know about ACRE.
ACE gunbag doesn't need to modify.

**Comment by jonpas** [comment]
ACRE has no radios in forms of backpacks, all are items (some can only fit a backpack), so no issue there.

**Comment by alganthe** [comment]
if you open the arsenal while you have a filled gunbag and switch to another backpack the content of the gunbag is cleared, that I can fix.

However saving the weapon that is inside the gunbag is not something I wish to do.
I'd need to add support for saving backpack variables, in which case you would be able to load non whitelisted weapons that are in gunbags.
I don't want that behavior.

**Comment by alganthe** [comment]
You can place weapons in vest and backpacks in ACE Arsenal, however they might be stripped out of their attachments and there's no command to add them back yet.

**Comment by DeliciousJaffa** [comment]
@alganthe Does ACE not use the setUnitLoadout and getUnitLoadout commands? They are the only commands that allow you to specify weapons inside of backpacks (with attachments).

I just tested in the Ace arsenal and you do appear to be using those commands as the weapon showed under misc items and saved in my loadout, however ACE's implementation seems to strip the accessories on loadout loading.

Exported loadout
```sqf
[
    ["arifle_MX_F","","acc_pointer_IR","optic_Aco",["30Rnd_65x39_caseless_mag",30],[],""],
    [],
    ["hgun_P07_F","","","",["16Rnd_9x21_Mag",17],[],""],
    ["U_B_CombatUniform_mcam",[["ACE_fieldDressing",2],["ACE_morphine",1],["30Rnd_65x39_caseless_mag",2,30]]],
    ["V_PlateCarrier1_rgr",[["30Rnd_65x39_caseless_mag",7,30],["16Rnd_9x21_Mag",2,17],["SmokeShell",1,1],["SmokeShellGreen",1,1],["Chemlight_green",2,1],["HandGrenade",2,1]]],
    ["B_Kitbag_mcamo",[[["arifle_MX_F","","acc_pointer_IR","optic_Aco",[],[],""],1],[["hgun_P07_F","","","",[],[],""],1]]],
    "H_HelmetB",
    "G_WirelessEarpiece_F",
    [],
    ["ItemMap","","","ItemCompass","tf_microdagr","NVGoggles"]
]
```

**Comment by alganthe** [comment]
I know why it happens and that has nothing to do with this ticket.

**Comment by dedmen** [comment]
Solution would still be to just store variables, and restore if the backpack stays the same or a similar one.
https://github.com/michail-nikolaev/task-force-arma-3-radio/blob/master/addons/core/functions/events/fnc_onArsenal.sqf#L30
TFAR does that already for radio backpacks by just using the arsenal eventhandlers.

**Comment by alganthe** [comment]
That is already the case dedmen, see #6192 

The OP wants gunbag weapons to persist through loadout saves, I do not want that behavior.

**Comment by dedmen** [comment]
> The OP wants gunbag weapons to persist through loadout saves

took me 1.5 years to realise that.

**Comment by jonpas** [comment]
@alganthe is there API to allow that?

**Comment by alganthe** [comment]
Someone could use the loadout save / load EHs for that since the name of the loadouts are unique.

**Comment by LinkIsGrim** [comment]
Fixed in #8923.
